# Determining *cif* flanking regions

Once *cifA* and/or *cifB* were confidently identified, the nature of the flanking genes upstream and downstream was investigated. We aimed to identify the local environment of the *cif* genes at a small scale (the 5â€“6 closest neighboring genes), but also to categorize the environment at a broader scale (several dozen genes upstream and downstream) in order to determine the overall genomic context (prophage, RAGE, etc.). To do this, we used the amino acid sequences generated by Prokka (`.faa` files) of each bacterial genome in which *cifA-cifB* were identified. For each genome, the *.faa* files were compared against the RAGE and WO databases created for this study (described in detail in `S5 Tables`, `S6 Table`, and `S7 Table`) using `OrthoFinder v2.5.4` (<https://github.com/davidemms/OrthoFinder>, doi: <https://10.1186/s13059-019-1832-y>) with the `blast` option. 

Note that the genes within each database encompass various types and often include multiple variants of the same gene. For instance, within RAGE elements, several versions of *TraA*, *TrAI*, *TraB*, *raD*, SpoT hydrolase genes, etc. can be found. Similarly, the WO prophage databases contain multiple variants of tail, head, replicationbaseplate-related genes, etc.

The command used was the same as in part. 3, and identicaly, as this step can be resource-intensive, we typically processed a maximum of 20 genomes per analysis:

```
orthofinder -f ./OrthoFinder_environment/ -t 4 -S blast 
```
where `OrthoFinder_environment` is a directory containing the corresponding `.faa` files of each genomes generated by Prokka and the *RAGE* or *WO* databases (`.fasta` files).

The `Orthogroups.tsv` file of each analyses shows the protein(s) in each genome that match the query sequences (grouped within the same "orthogroup"), ie. the genes of the database. This method provides information of the type of genes and funtions of their associated protein located upstream and downstream of *cifA-cifB*. These new informations were reported on a consensus Table comprising key information about syntenic gene environments bordering *cifA-cifB*. An example of this table, named `Environment.txt`, is provided for two typical *cifA-cifB* environments: RAGE and WO prophages. eg. (i) *Rickettsia hoogstraalii* strain CS carry *cif* operon in a RAGE module on contig29, and (ii) *Mesenet longicola* strain GL2 carry *cif* operon in a WO prophage or WO-like island on contig9. 
The `Environment.txt` table indicates the distinct ORFs identified by Prokka ('Prokka_name') and their correspondance ('Gene') in the RAGE/WO databases or BlastP (see below). Regarding the two examples, contig29 of *R. hoogstraalii* CS contains 92 ORFs according to Prokka (only 56 are presented in the `Environment.txt` table); contig9 of *M. lonficola* GL2 hold 22 ORFs (all presented in `Environment.txt`).

Note that in the *cifA-cifB* genomic environments of each genome, there were always several genes that did not match any entry in the RAGE or WO databases. In cases where the environment did not correspond to either a RAGE or a WO element, none of the surrounding genes matched. For these cases, the unidentified genes were compared against the NCBI database using BlastP (<https://blast.ncbi.nlm.nih.gov/Blast.cgi>). Functional annotations were assigned to genes showing >70% query coverage and percentage identity.
