# Determining *cif* flanking regions

Once *cifA* and/or *cifB* were confidently identified, the nature of the flanking genes upstream and downstream was investigated. We aimed to identify the local environment of the *cif* genes at a small scale (the 5â€“6 closest neighboring genes), but also to categorize the environment at a broader scale (several dozen genes upstream and downstream) in order to determine the overall genomic context (prophage, RAGE, etc.). To do this, we used the amino acid sequences (`.faa` files) of each bacterial genome in which *cifA-cifB* were identified. For each genome, the *.faa* files were compared against the RAGE and WO databases created for this study (described in detail in `S5 Tables`, `S6 Table`, and `S7 Table`) using `OrthoFinder v2.5.4` (<https://github.com/davidemms/OrthoFinder>, doi: <https://10.1186/s13059-019-1832-y>) with the `blast` option. The command used was the same as in part 3, and identicaly, as this step can be resource-intensive, we typically processed a maximum of 20 genomes per analysis:

```
orthofinder -f ./OrthoFinder_environment/ -t 4 -S blast 
```
where `OrthoFinder_environment` is a directory containing the corresponding `.faa` files of each genomes generated by Prokka and the *RAGE* or *WO* databases (`.fasta` files).

Note that the genes within each database encompass various types and often include multiple variants of the same gene. For instance, within RAGE elements, several versions of *TraA*, *TrAI*, *TraB*, *raD*, SpoT hydrolase genes, etc. can be found. Similarly, the WO prophage databases contain multiple variants of tail, head, replicationbaseplate-related genes, etc.

The `Orthogroups.tsv` file of each analyses shows the protein(s) in each genome that match the query sequences (grouped within the same "orthogroup"), ie. the genes of the database. This method provides information of the type of genes and funtions of their associated protein located upstream and downstream of cifA-cifB. 
